File: frontend/src/app/interfaces/EntregaElaboracionViewModel.ts
import ElaboracionViewModel from "./ElaboracionViewModel";
import InsumoViewModel from "./InsumoViewModel";
import { PuntoVentaViewModel } from "./PuntoVentaViewModel";
import { UsuarioViewModel } from "./UsuarioViewModel";

export interface EntregaElaboracionViewModel {
  id: number;
  cantidad: number;
  fecha: Date;
  autor: UsuarioViewModel;
  elaboracion: ElaboracionViewModel;
  puntoVenta: PuntoVentaViewModel;
}

File: models/src/main/java/com/app/viewModels/EntregaElaboracionViewModel.java
package com.app.viewModels;


import com.app.viewModels.base.IdentifiableViewModel;

import java.util.Date;

public class EntregaElaboracionViewModel extends IdentifiableViewModel {

    public long cantidad;
    public ElaboracionViewModel elaboracion;
    public PuntoVentaViewModel puntoVenta;
    public Date fecha;
    public UsuarioViewModel autor;

    public EntregaElaboracionViewModel(long cantidad, ElaboracionViewModel elaboracion, Date fecha, PuntoVentaViewModel puntoVenta) {
        this.cantidad = cantidad;
        this.elaboracion = elaboracion;
        this.fecha = fecha;
        this.puntoVenta = puntoVenta;
    }

    public EntregaElaboracionViewModel(Long id, long cantidad, ElaboracionViewModel elaboracion, Date fecha, PuntoVentaViewModel puntoVenta, UsuarioViewModel autor) {
        super(id);
        this.cantidad = cantidad;
        this.elaboracion = elaboracion;
        this.fecha = fecha;
        this.puntoVenta = puntoVenta;
        this.autor = autor;
    }
}

